### DISCUSSION TITLE
How do I filter the API by court ID?

### DISCUSSION THREAD

**Discussion by text2sql** [freelawproject/courtlistener#3215] [Q&A] [answered]
I am trying to find precedential decisions by 'il' court after 2023-01-01 with this api query https://www.courtlistener.com/api/rest/v3/opinions/?cluster__date_filed__gte=2023-01-01&cluster__docket__court_id=il&cluster__precedential_status=Published but am getting Texas courts instead :-) 
The web interface worked fine 
https://www.courtlistener.com/?type=o&q=&type=o&order_by=score desc&stat_Precedential=on&filed_after=01%2F01%2F2023&court=ill 
Where did I go wrong on the API query? 
Thank you

**Comment by mlissner** [comment, marked as answer]
Well, invalid fields will be ignored, so if you're getting Texas content when you try to filter to Illinois, that means your filter didn't work (and probably Texas is just at the top of whatever returned). 

Your query has two problems:

1. You didn't filter to the court properly.
2. You didn't use the correct court (it's `ill`, not `il`).

So you should do this instead:

https://www.courtlistener.com/api/rest/v3/opinions/?cluster__date_filed__gte=2023-01-01&cluster__docket__court=ill&cluster__precedential_status=Published

You could also do this:

https://www.courtlistener.com/api/rest/v3/opinions/?cluster__date_filed__gte=2023-01-01&cluster__docket__court__id=ill&cluster__precedential_status=Published

But that involves another join, so it's slower.

**Reply by text2sql** [reply]
thank you, it worked! so, going forward, when searching by courts, better to use use court= and not court__id?

**Reply by mlissner** [reply]
Yes. The latter does an extra join.

**Comment by mlissner** [comment]
I've cleaned up this answer and added it as an FAQ on our new wiki: https://wiki.free.law/c/courtlistener/help/api/faqs/how-do-i-filter-the-case-law-apis-by-court
